packages/turf-standard-deviational-ellipse/index.js

```diff
@@ -0,0 +1,133 @@
+ *
+ * @name standardDeviationalEllipse
+ * @param {Geometry|FeatureCollection<Point>} points GeoJSON points
+ * @param {Object} [options={}] Optional parameters
+ * @param {string} [options.weight] the property name used to weight the center
+ * @param {number} [options.steps=64] number of steps for the polygon
+ * @param {Object} [options.properties={}] properties to pass to the resulting ellipse
+ * @returns {Feature<Polygon>} an elliptical Polygon that includes approximately 1 SD of the dataset within it.
+ * @example
+ *
+ * var bbox = [-74, 40.72, -73.98, 40.74];
+ * var points = turf.randomPoint(400, {bbox: bbox});
+ * var sdEllipse = turf.standardDeviationalEllipse(points, options);
+ *
+ * //addToMap
+ * var addToMap = [points, sdEllipse];
+ *
+ */
+function standardDeviationalEllipse(points, options) {
+    // Optional params
+    options = options || {};
+    if (!isObject(options)) throw new Error('options is invalid');
+    var steps = options.steps || 64;
+    var weightTerm = options.weight;
+    var properties = options.properties || {};
+
+    // Validation:
+    if (!isNumber(steps)) throw new Error('steps must be a number');
+    if (!isObject(properties)) throw new Error('properties must be a number');
+
+    // Calculate mean center & number of features:
+    var numberOfFeatures = coordAll(points).length;
+    var meanCenter = centerMean(points, {weight: weightTerm});
+
+    // Calculate angle of rotation:
+    // [X, Y] = mean center of all [x, y].
+    // theta = arctan( (A + B) / C )
+    // A = sum((x - X)^2) - sum((y - Y)^2)
+    // B = sqrt(A^2 + 4(sum((x - X)(y - Y))^2))
+    // C = 2(sum((x - X)(y - Y)))
+
+    var xDeviationSquaredSum = 0;
+    var yDeviationSquaredSum = 0;
+    var xyDeviationSum = 0;
+
+    featureEach(points, function (point) {
+        var weight = point.properties[weightTerm] || 1;
+        var deviation = getDeviations(getCoords(point), getCoords(meanCenter));
+        xDeviationSquaredSum += Math.pow(deviation.x, 2) * weight;
+        yDeviationSquaredSum += Math.pow(deviation.y, 2) * weight;
+        xyDeviationSum += deviation.x * deviation.y * weight;
+    });
+
+    var bigA = xDeviationSquaredSum - yDeviationSquaredSum;
+    var bigB = Math.sqrt(Math.pow(bigA, 2) + 4 * Math.pow(xyDeviationSum, 2));
+    var bigC = 2 * xyDeviationSum;
+    var theta = Math.atan((bigA + bigB) / bigC);
+    var thetaDeg = theta * 180 / Math.PI;
+
+    // Calculate axes:
+    // sigmaX = sqrt((1 / n - 2) * sum((((x - X) * cos(theta)) - ((y - Y) * sin(theta)))^2))
+    // sigmaY = sqrt((1 / n - 2) * sum((((x - X) * sin(theta)) - ((y - Y) * cos(theta)))^2))
+    var sigmaXsum = 0;
+    var sigmaYsum = 0;
+    var weightsum = 0;
+    featureEach(points, function (point) {
```

Comment:
@muziejus I might be off on this next comment, but is there any reason why we need to iterate over the Points twice? Does this 2nd iteration need the sum values of the first pass?

Reply:
The first iteration gets us theta, which we need for the second iteration. I tried to factor them together, but, alas.

Reply:
👍 that's what I thought